projects/systelab-components/src/lib/contextmenu/context-menu-submenu-item.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, ElementRef, Input, QueryList, ViewChildren, } from '@angular/core';
+import { ContextMenuOption } from './context-menu-option';
+import { ContextMenuComponent } from './context-menu.component';
+
+@Component({
+	selector:    'systelab-context-menu-submenu-item',
+	templateUrl: 'context-menu-submenu-item.component.html'
+})
+export class ContextMenuSubmenuItemComponent {
+
+	@Input() public action: ContextMenuOption;
+	@Input() public actionName: string;
+	@Input() public hasIcons: boolean;
+	@Input() public hasChildren: boolean;
+	public readonly levelSeparator = '_|_';
+	@Input() public contextMenuOriginal: ContextMenuComponent;
+	@Input() public parentId: string;
+
+	@Input() public elementID = (Math.floor(Math.random() * (999999999999 - 1))).toString();
+
+	@ViewChildren('childdropdownmenu') public childDropdownMenuElement: QueryList<ElementRef>;
+
+	constructor() {
+	}
+
+	public doMouseOver(event: any, elementID: string, actionId: string): void {
+		let selectedChild;
+
+		const optionAcitionId = this.contextMenuOriginal.getOptionDetailsActionId(actionId);
+
+		selectedChild = this.childDropdownMenuElement.toArray()
+			.find((elem) => elem.nativeElement.id === (optionAcitionId + this.elementID));
+
+		this.contextMenuOriginal.showSubmenu(event, actionId, selectedChild, this.elementID);
+	}
+
+	public doClickWithString(event: any, elementID: string, actionId: string): void {
```

Comment:
withString.: Avoid use the type of a parameter for the name of a method

Reply:
Done